airflow/api/client/local_client.py

```diff
@@ -28,15 +28,32 @@
 class Client(api_client.Client):
     """Local API client implementation."""
 
-    def trigger_dag(self, dag_id, run_id=None, conf=None, execution_date=None, replace_microseconds=True):
+    def trigger_dag(
+        self, dag_id, run_id=None, conf=None, execution_date=None, replace_microseconds=True
+    ) -> dict | None:
         dag_run = trigger_dag.trigger_dag(
             dag_id=dag_id,
             run_id=run_id,
             conf=conf,
             execution_date=execution_date,
             replace_microseconds=replace_microseconds,
         )
-        return f"Created {dag_run}"
+        if dag_run:
+            return {
+                "conf": dag_run.conf,
+                "dag_id": dag_run.dag_id,
+                "dag_run_id": dag_run.run_id,
+                "data_interval_end": dag_run.data_interval_start,
```

Comment:
what's the rational behind the swapping of  `data_interval_end -> dag_run.data_interval_start` and `data_interval_start -> dag_run.data_interval_end` here ?

Reply:
Clearly this is a bug, here is the fix: #32548